### PR TITLE
1036 check file limit after datsource selection

### PR DESCRIPTION
**Pull request by millmanorama**

### CODE REVIEW

File: ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryTopComponent.java [hunk 1]
```diff
@@ -147,82 +152,139 @@ public static TopComponent getTopComponent() {
...
+        new Thread(new Task<Void>() {
+            @Override
+            protected Void call() throws Exception {
+
+                List<DataSource> dataSources = new ArrayList<>();
+                dataSources.addAll(controller.getSleuthKitCase().getDataSources());
+
+                /*
+                 * If there is only one datasource or the grouping is already
+                 * set to something other than path , don't both to ask for
```
Comment (raman-bt) [inline ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryTopComponent.java]: type in comment - "bother"

File: ImageGallery/src/org/sleuthkit/autopsy/imagegallery/gui/DataSourceCell.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package org.sleuthkit.autopsy.imagegallery.gui;
```
Comment (raman-bt) [inline ImageGallery/src/org/sleuthkit/autopsy/imagegallery/gui/DataSourceCell.java]: Add a header

File: ImageGallery/src/org/sleuthkit/autopsy/imagegallery/gui/Toolbar.java [hunk 3]
```diff
@@ -316,12 +320,24 @@ public void onFailure(Throwable t) {
 
     @ThreadConfined(type = ThreadConfined.ThreadType.ANY)
     private ListenableFuture<List<DataSource>> syncDataSources() {
-        ListenableFuture<List<DataSource>> future = exec.submit(controller.getSleuthKitCase()::getDataSources);
+        ListenableFuture<List<DataSource>> future = exec.submit(() -> {
+            List<DataSource> dataSources1 = controller.getSleuthKitCase().getDataSources();
```
Comment (raman-bt) [inline ImageGallery/src/org/sleuthkit/autopsy/imagegallery/gui/Toolbar.java]: Can 'dataSources1' be named better?

**Comment by raman-bt** [comment]
I've tested this PR and approve with the minor comments above.

**Comment by millmanorama** [comment]
I addressed @raman-bt's comments and fixed some codacy issues as well

**Pull request merged**